server/benches/bench.rs

```diff
@@ -0,0 +1,184 @@
+    let setup = Rc::new(Mutex::new(setup));
+
+    let mut seed = 42;
+    let mut rng = || {
+        seed = rand(seed);
+        seed
+    };
+
+    let alice: QualifiedUserName = format!("alice{}@example.com", rng()).parse().unwrap();
+    let bob: QualifiedUserName = format!("bob{}@example.com", rng()).parse().unwrap();
+
+    let conversation_alice_bob = runtime.block_on(async {
+        let mut setup = setup.lock().await;
+        setup.add_user(&alice).await;
+        setup.add_user(&bob).await;
+        setup.connect_users(&alice, &bob).await
+    });
+
+    let mut group = c.benchmark_group("benchmarks");
+
+    // We reduce the sample size to make `invite_to_group` pass. Otherwise, we run out of key packages.
+    group.sample_size(30);
+
+    group.bench_function("add_user", |b| {
+        b.to_async(&runtime).iter_custom(|iter| {
+            let suffix = rng();
+            let setup = setup.clone();
+            let mut elapsed = Duration::default();
+            async move {
+                let mut setup = setup.lock().await;
+                for i in 0..iter {
+                    let bob: QualifiedUserName =
+                        format!("bob_{i}_{suffix}@example.com").parse().unwrap();
+                    let time = Instant::now();
+                    setup.add_user(&bob).await;
+                    elapsed += time.elapsed();
+                }
+                elapsed
+            }
+        })
+    });
+
+    group.bench_function("connect_users", |b| {
+        b.to_async(&runtime).iter_custom(|iter| {
+            let alice = alice.clone();
+            let suffix = rng();
+            let setup = setup.clone();
+            let mut elapsed = Duration::default();
+            async move {
+                let mut setup = setup.lock().await;
+                for i in 0..iter {
+                    let bob: QualifiedUserName =
+                        format!("bob_{i}_{suffix}@example.com").parse().unwrap();
+                    setup.add_user(&bob).await;
+                    let time = Instant::now();
+                    setup.connect_users(&alice, &bob).await;
+                    elapsed += time.elapsed();
+                }
+                elapsed
+            }
+        })
+    });
+
+    group.bench_function("send_message", |b| {
+        b.to_async(&runtime).iter_custom(|iter| {
+            let setup = setup.clone();
+            let alice = alice.clone();
+            let bob = bob.clone();
+            let mut elapsed = Duration::default();
+
+            async move {
+                let mut setup = setup.lock().await;
+                for _ in 0..iter {
+                    let time = Instant::now();
+                    setup
+                        .send_message(conversation_alice_bob, &alice, vec![&bob])
+                        .await;
+                    elapsed += time.elapsed()
+                }
+                // update group, otherwise we get too far in to the future error from MLS
```

Comment:
Can you file an issue for this? The test framework should be able to handle that